<launch>
    <arg name="veh" doc="Name of vehicle. ex: megaman"/>
    <arg name="pkg_name" value="deadreckoning"/>
    <arg name="node_name" default="deadreckoning_node"/>
    <arg name="param_file_name" default="default" doc="Specify a param file. ex: experiment_1."/>


    <group ns="$(arg veh)">
        <remap from="deadreckoning_node/left_wheel" to="left_wheel_encoder_node/tick"/>
        <remap from="deadreckoning_node/right_wheel" to="right_wheel_encoder_node/tick"/>
        <node name="$(arg node_name)" pkg="$(arg pkg_name)" type="$(arg node_name).py" output="screen" required="true">
            <param name="veh" value="$(arg veh)" />
            <rosparam command="load"
                file="$(find deadreckoning)/config/$(arg node_name)/$(arg param_file_name).yaml"/>
        </node>
    </group>
</launch>
